backend/src/main/kotlin/org/loculus/backend/service/submission/FileMappingPreconditionValidator.kt

```diff
@@ -70,13 +71,40 @@ class FileMappingPreconditionValidator(
         return this
     }
 
-    fun validateFilesExist(fileIds: Set<FileId>): FileMappingPreconditionValidator {
+    fun validateFileIdsExist(fileIds: Set<FileId>): FileMappingPreconditionValidator {
+        val nonExistentFileIds = filesDatabaseService.getNonExistentFileIds(fileIds)
+        if (nonExistentFileIds.isNotEmpty()) {
+            throw UnprocessableEntityException(
+                "The following file IDs do not exist: " + nonExistentFileIds.joinToString(),
+            )
+        }
+        return this
+    }
+
+    fun validateFilesUploadedNotJustRequested(fileIds: Set<FileId>): FileMappingPreconditionValidator {
```

Comment:
```suggestion
    fun validateFilesUploaded(fileIds: Set<FileId>): FileMappingPreconditionValidator {
```

IMO this is over-long and the extra words don't really add clarity
(not blocking)